Update to 1.21.3

bl4ckscor3 opened the pull request:
Many events that summon mobs were affected by a base game change. I switched all the spawn reasons to mob summoned, matching vexes being spawned by evokers. The other big change are post process effect definitions.

Not working: CRT and LSD shader (visuals don't match between versions); Herobrine fog

slicedlime commented on src/main/java/me/juancarloscp52/entropy/events/db/AngryBeeEvent.java:
```diff
@@ -17,7 +17,7 @@ public void init() {
                 serverPlayerEntity -> {
                     for (int i = 0; i < 3; i++) {
-                        Bee bee = EntityType.BEE.spawn(serverPlayerEntity.serverLevel(),serverPlayerEntity.blockPosition().east(2), MobSpawnType.SPAWN_EGG);
+                        Bee bee = EntityType.BEE.spawn(serverPlayerEntity.serverLevel(),serverPlayerEntity.blockPosition().east(2), EntitySpawnReason.MOB_SUMMONED);
```
I'd argue `COMMAND` or `EVENT` would be a more accurate reason.

slicedlime commented on src/main/java/me/juancarloscp52/entropy/events/db/MidasTouchEvent.java:
```diff
@@ -29,6 +30,8 @@
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.block.Blocks;
 import net.minecraft.world.phys.AABB;
+
+import java.awt.*;
```
Unused.

slicedlime commented on src/main/java/me/juancarloscp52/entropy/mixin/BlockMixin.java:
```diff
@@ -51,7 +52,7 @@ private static void randomDrops(Level world, BlockPos pos, ItemStack stack, Call
         }
         if (Variables.randomDrops || Variables.luckyDrops) {
-            if (!world.isClientSide && !stack.isEmpty() && world.getGameRules().getBoolean(GameRules.RULE_DOBLOCKDROPS)) {
+            if (!world.isClientSide && !stack.isEmpty() && ((ServerLevel) world).getGameRules().getBoolean(GameRules.RULE_DOBLOCKDROPS)) {
```
```suggestion
            if (world instanceof ServerLevel serverLevel && !stack.isEmpty() && serverLevel.getGameRules().getBoolean(GameRules.RULE_DOBLOCKDROPS)) {
```

slicedlime reviewed (commented):
Looked through it - code looks good to me this far at least.

bl4ckscor3 commented:
Fixed CRT and Herobrine events - the Herobrine event did not match its initial implementation from 1.16.5 since at least 1.18.2, as the fog had started to be way too far out. I fixed that. Additionally, while the fog in Entropy for 1.16.5 was white, I changed it to black to add to the eeriness.

slicedlime commented on src/main/java/me/juancarloscp52/entropy/mixin/FogRendererMixin.java:
```diff
@@ -17,27 +17,23 @@
-    @Inject(method = "setupFog", at = @At(value = "RETURN", ordinal = 1))
-    private static void changeFogDistance(Camera camera, FogRenderer.FogMode fogMode, Vector4f fogColor, float renderDistance, boolean isFoggy, float partialTick, CallbackInfoReturnable<FogParameters> cir, @Local FogRenderer.FogData fogData) {
+    @ModifyReturnValue(method = "setupFog", at = @At(value = "RETURN", ordinal = 1))
+    private static FogParameters changeFogDistance(FogParameters original) {
```
This means the accesswidener is no longer needed.
